tools/depcheck/pkg/cmd/trace/trace.go

```diff
@@ -0,0 +1,301 @@
+package trace
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"strings"
+
+	"github.com/spf13/cobra"
+
+	"github.com/gonum/graph"
+	"github.com/gonum/graph/encoding/dot"
+
+	depgraph "github.com/openshift/origin/tools/depcheck/pkg/graph"
+)
+
+var traceImportsExample = `# create a dependency graph
+%[1]s trace --root=pkg/one --root=pkg/two
+
+# create a dependency graph and output in "dot" format
+%[1]s trace --root=pkg/one --output=dot
+`
+
+type TraceImportsOpts struct {
+	Roots   []string
+	BaseDir string
+
+	excludes []string
+	filters  []string
+
+	Analyze          string
+	ExcludeFileName  string
+	CollapseFileName string
+
+	OutputFormat string
+
+	Out    io.Writer
+	ErrOut io.Writer
+}
+
+type TraceImportsFlags struct {
+	Roots        []string
+	OutputFormat string
+	Exclude      string
+	Collapse     string
+	Analyze      string
+}
+
+func (o *TraceImportsFlags) ToOptions(out, errout io.Writer) (*TraceImportsOpts, error) {
+	return &TraceImportsOpts{
+		Roots:            o.Roots,
+		ExcludeFileName:  o.Exclude,
+		CollapseFileName: o.Collapse,
+		Analyze:          o.Analyze,
+
+		OutputFormat: o.OutputFormat,
+
+		Out:    out,
+		ErrOut: errout,
+	}, nil
+}
+
+func NewCmdTraceImports(parent string, out, errout io.Writer) *cobra.Command {
+	flags := &TraceImportsFlags{}
+
+	cmd := &cobra.Command{
+		Use:     "trace --root=pkg/one",
+		Short:   "Creates and analyzes a dependency graph from a set of given root packages",
+		Long:    "Creates and analyzes a dependency graph from a set of given root packages",
+		Example: fmt.Sprintf(traceImportsExample, parent),
+		RunE: func(c *cobra.Command, args []string) error {
+			opts, err := flags.ToOptions(out, errout)
+			if err != nil {
+				return err
+			}
+
+			if err := opts.Complete(); err != nil {
+				return err
+			}
+			if err := opts.Validate(); err != nil {
+				return err
+			}
+			if err := opts.Run(); err != nil {
+				return err
+			}
+
+			return nil
+		},
+	}
+
+	cmd.Flags().StringSliceVar(&flags.Roots, "root", flags.Roots, "set of entrypoints for dependency trees used to generate a depedency graph.")
+	cmd.Flags().StringVarP(&flags.Exclude, "exclude", "e", "", "json file containing a list of import-paths of packages to recursively exclude when traversing the set of given entrypoints specified through --root.")
```

Comment:
description sounds bad.  I'd want to choose recursive exclusion or not based on `...` in golang, right?